### PR TITLE
Fix content share resume after reconnection

### PR DESCRIPTION
**Pull request by linsang21**
## ℹ️ Description
* Fix the issue that the screen share doesn't resume after reconnection
* Update demo build without bitcode by default since it's been ignore any way
* Update Podfile to use latest SDKs for easy set up

### Issue #, if available
P109455384

### Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
    - [x] README update
    - [x] CHANGELOG update
    - [ ] guides update
- [ ] This change requires a dependency update
    - [ ] Amazon Chime SDK Media
    - [ ] Other (update corresponding legal documents)

## 🧪 How Has This Been Tested?
* Build demo with the Podfile, verified it worked
* Made change in SDK and demo, verify the screen share can resume after reconnection

### Additional Manual Test
- [ ] Pause and resume remote video
- [ ] Switch local camera
- [ ] Rotate screen back and forth

## 📱 Screenshots, if available
*provide screenshots/video record if there's a UI change in demo app*

#### Integration validation (required before release)

- [ ] Unit tests pass
- [ ] Build SDK against simulator with release options
- [ ] Build SDK against device with release options
- [ ] Build SDK against simulator with debug options
- [ ] Build SDK against device with debug options
- [ ] Test Demo against simulator with SDK (debug build) in workspace
- [ ] Test Demo against device with SDK (debug build) in workspace
- [ ] Test DemoObjC against simulator with SDK (debug build) in workspace
- [ ] Test DemoObjC against device with SDK (debug build) in workspace
- [ ] Test Demo against simulator with SDK.framework (release build)
- [ ] Test Demo against device with SDK.framework (release build)
- [ ] Test DemoObjC against simulator with SDK.framework (release build)
- [ ] Test DemoObjC against device with SDK.framework (release build)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: AmazonChimeSDK/AmazonChimeSDK/internal/video/VideoClientProtocol.swift [hunk 1]
```diff
@@ -11,7 +11,7 @@ import AmazonChimeSDKMedia
 import Foundation
 
 @objc public protocol VideoClientProtocol {
-    var delegate: VideoClientDelegate! { get set }
+    weak var delegate: VideoClientDelegate! { get set }
```
Comment (georgezy-amzn) [inline AmazonChimeSDK/AmazonChimeSDK/internal/video/VideoClientProtocol.swift:14]: Also avoid force unwrapping the delegate:  `VideoClientDelegate?`

File: AmazonChimeSDKDemo/Podfile [hunk 2]
```diff
@@ -4,12 +4,12 @@ platform :ios, '11.0'
 target 'AmazonChimeSDKDemoPods' do
   use_frameworks!
   # Pods for AmazonChimeSDKDemoPods
-  pod 'AmazonChimeSDK-Bitcode', 'AMAZON_CHIME_SDK_VERSION'
```
Comment (georgezy-amzn) [inline AmazonChimeSDKDemo/Podfile:7]: Won't this break Jenkins? Iirc, those are the placeholders for the target SDK version to build.

**Review by georgezy-amzn — COMMENTED** [review]
It's good intention to maintain the dependencies, but I'd recommend to address them in separate PRs.

File: AmazonChimeSDK/AmazonChimeSDK/internal/video/VideoClientProtocol.swift (hunk 1 above)
Comment (linsang21) [reply AmazonChimeSDK/AmazonChimeSDK/internal/video/VideoClientProtocol.swift:14]: Sure

File: AmazonChimeSDKDemo/Podfile (hunk 2 above)
Comment (linsang21) [reply AmazonChimeSDKDemo/Podfile:7]: Sigh. OK I'll get this part out of this PR.

File: AmazonChimeSDK/AmazonChimeSDK/internal/video/VideoClientProtocol.swift (hunk 1 above)
Comment (linsang21) [reply AmazonChimeSDK/AmazonChimeSDK/internal/video/VideoClientProtocol.swift]: Discussed offline, this is an improvement on fixing memory leak, which will be handled separately. Updated the code to still use the old way and add a TODO.

File: AmazonChimeSDK/AmazonChimeSDK/internal/video/VideoClientProtocol.swift [hunk 3]
```diff
@@ -11,6 +11,7 @@ import AmazonChimeSDKMedia
 import Foundation
 
 @objc public protocol VideoClientProtocol {
+    // TODO: look into make delegate weak preference
```
Comment (georgezy-amzn) [inline AmazonChimeSDK/AmazonChimeSDK/internal/video/VideoClientProtocol.swift:14]: Typo: weak reference

**Review by georgezy-amzn — APPROVED** [review]

**Review by dylonChime — APPROVED** [review]

**Pull request merged**